source/verus/src/main.rs

```diff
@@ -75,19 +63,270 @@ fn main() {
+        let job = win32job::Job::create()
+            .map_err(|_| format!("last os error: {}", std::io::Error::last_os_error()))?;
+        let mut info = job
+            .query_extended_limit_info()
+            .map_err(|_| format!("last os error: {}", std::io::Error::last_os_error()))?;
+        info.limit_kill_on_job_close();
+        job.set_extended_limit_info(&mut info)
+            .map_err(|_| format!("last os error: {}", std::io::Error::last_os_error()))?;
+        job.assign_current_process()
+            .map_err(|_| format!("last os error: {}", std::io::Error::last_os_error()))?;
+        std::mem::forget(job);
+    }
+
+    let reports = match reports {
+        Some(reports) => {
+            // proj path is only created after the project has --local-store flag
+            if !reports.proj_path.is_dir() {
+                cmd.status().map_err(|x| format!("verus failed to run with error: {}", x))?;
+                return Ok(());
+            }
+            reports
+        }
+        None => {
+            cmd.status().map_err(|x| format!("verus failed to run with error: {}", x))?;
+            return Ok(());
+        }
+    };
+
+    let proj_path = reports.proj_path;
+
+    cmd.arg(format!("--emit=dep-info={}", reports.dep_path.display()));
+
+    // clean all files in proj_path
+    remove_dir_all(&proj_path).expect("failed to remove repo_path");
+
+    let toml_path = proj_path.join("reports.toml");
+
+    create_dir_all(&proj_path).expect("creating reports.toml");
+    let mut file = File::create(toml_path).expect("creating reports.toml");
+
+    let mut child = cmd
+        .stdout(std::process::Stdio::piped())
+        .stderr(std::process::Stdio::piped())
+        .spawn()
+        .map_err(|x| format!("verus failed to run with error: {}", x))?;
+
+    let out = BufReader::new(child.stdout.take().expect("create BufReader for stdout"));
+    let err = BufReader::new(child.stderr.take().expect("create BufReader for stderr"));
+
+    let thread_err = std::thread::spawn(move || {
+        let mut stderr_output = String::new();
+        err.lines().for_each(|line| {
+            let line = line.unwrap();
+            eprintln!("{}", line);
+            let x = anstream::adapter::strip_str(&line).to_string() + "\n";
+            stderr_output.push_str(&x);
+        });
+        return stderr_output;
+    });
+
+    let thread_out = std::thread::spawn(move || {
+        let mut stdout_output = String::new();
+        out.lines().for_each(|line| {
+            let line = line.unwrap();
+            println!("{}", line);
+            let x = anstream::adapter::strip_str(&line).to_string() + "\n";
+            stdout_output.push_str(&x);
+        });
+        return stdout_output;
+    });
+
+    let err_string = thread_err.join().unwrap();
+    let out_string = thread_out.join().unwrap();
+
+    child.wait().map_err(|x| format!("verus failed to run with error: {}", x))?;
```

Comment:
You want to capture the resulting `ExitCode` you get as an output into a variable, and then pass it along to the calling function so that the exit code is handled correctly. Currently it always exits with `0` irrespective of how the child exited, which is incorrect.